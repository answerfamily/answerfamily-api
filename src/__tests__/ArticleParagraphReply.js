const {
  loadESFixtures,
  unloadESFixtures,
  loadMongoFixtures,
  unloadMongoFixtures,
} = require('../../test/fixtures');
const gql = require('../../test/gql');
const {
  ES_FIXTURES,
  MONGO_FIXTURES,
} = require('../__fixtures__/articleParagraphReply');

describe('Basic object type graph resolving', () => {
  beforeAll(async () => {
    await loadESFixtures(ES_FIXTURES);
    await loadMongoFixtures(MONGO_FIXTURES);
  });

  afterAll(async () => {
    await unloadESFixtures(ES_FIXTURES);
    await unloadMongoFixtures(MONGO_FIXTURES);
  });

  it('loads from article to reply', async () => {
    const { data, errors } = await gql`
      {
        article(id: "article1") {
          text
          paragraphs {
            id
            text
            paragraphReplies {
              id
              reply {
                id
                text
              }
              canDelete
            }
            canDelete
          }
          createdAt
          sources {
            id
            article {
              id
            }
            url
            hyperlink {
              title
            }
          }
          hyperlinks {
            title
          }
        }
      }
    `({}, { userPromise: Promise.resolve({ iss: 'other-author' }) });

    expect(errors).toBeUndefined();
    expect(data).toMatchSnapshot();
  });

  it('loads from reply to article', async () => {
    const { data, errors } = await gql`
      {
        reply(id: "r1") {
          paragraphReplies {
            id
            paragraph {
              id
              article {
                id
              }
              canDelete
            }
            canDelete
          }
          hyperlinks {
            title
          }
        }
      }
    `(
      {},
      {
        userPromise: Promise.resolve({ iss: 'author1' }),
      }
    );

    expect(errors).toBeUndefined();
    expect(data).toMatchSnapshot();
  });

  it('loads from paragraph as well', async () => {
    const { data, errors } = await gql`
      {
        paragraph(id: "p1") {
          article {
            id
          }
          paragraphReplies {
            id
            reply {
              id
              text
            }
          }
        }
      }
    `();

    expect(errors).toBeUndefined();
    expect(data).toMatchSnapshot();
  });
});
