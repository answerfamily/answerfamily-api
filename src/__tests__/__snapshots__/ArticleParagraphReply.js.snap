// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic object type graph resolving loads from article to reply 1`] = `
Object {
  "article": Object {
    "createdAt": "2018-10-29T02:59:39.039Z",
    "hyperlinks": Array [
      Object {
        "title": "國二健教課本惹議 竟教性滿足、自拍性愛－民視新聞",
      },
    ],
    "paragraphs": Array [
      Object {
        "canDelete": false,
        "id": "p1",
        "paragraphReplies": Array [
          Object {
            "canDelete": false,
            "id": "5bd677bb092a3ee2541b1a0d",
            "reply": Object {
              "id": "r1",
              "text": "
      「外國同性戀藉結婚來台治療愛滋病，都是健保支付，全民買單」這句話其實並不算錯，但實務上並不會有「拖垮健保」的狀況，也跟同性婚姻關係不大，所以不用太擔心唷。
      衛福部 - 有關網路流傳外國人染愛滋病來台2年後，所有外國人的愛滋病治療費用是由健保署負擔，疾管署特此說明
      https://www.cdc.gov.tw/info.aspx?treeid=45da8e73a81d495d&nowtreeid=1bd193ed6dabaee6&tid=14D38681253864AB&fbclid=IwAR1mtdGIYYbHdiB99pZg_YQSwGXUD-DMj47fmgFnrphQ778k73SQzZkEUDk
    ",
            },
          },
        ],
        "text": "外國同性戀藉結婚來台治療愛滋病，都是健保支付，全民買單",
      },
      Object {
        "canDelete": false,
        "id": "p2",
        "paragraphReplies": Array [
          Object {
            "canDelete": false,
            "id": "5bd677ef092a3ee2541b1a1d",
            "reply": Object {
              "id": "r2",
              "text": "
      國中小的「同志教育」其實一直都在《性別平等教育法實行細則》裡頭，只是在保守團體的打壓之下，在國小的課本中沒有出現過。
      《同志教育怎麼教？》（含字幕）https://www.youtube.com/watch?v=mp2CCtEPAPA
    ",
            },
          },
        ],
        "text": "學校對國中小實施同志教育，將成為合法",
      },
    ],
    "sources": Array [
      Object {
        "article": Object {
          "id": "article1",
        },
        "hyperlink": Object {
          "title": "Google",
        },
        "id": "5bd677ef092a3ee2441b1a1d",
        "url": "http://google.com",
      },
    ],
    "text": "大法官已釋憲，如果年底公投沒達到500萬票，明年五月台灣將成為亞洲第一個同性可合法結婚的國家。

外國同性戀藉結婚來台治療愛滋病，都是健保支付，全民買單。

學校對國中小實施同志教育，將成為合法。

這次公投要主動向票務員領10、11、12三張公投單選 “同意”

11/24一定要去公投!讓我們盡一份心力保護後代子孫。 https://m.youtube.com/watch?v=ZXuUuVtc07Q&feature=youtu.be",
  },
}
`;

exports[`Basic object type graph resolving loads from paragraph as well 1`] = `
Object {
  "paragraph": Object {
    "article": Object {
      "id": "article1",
    },
    "paragraphReplies": Array [
      Object {
        "id": "5bd677bb092a3ee2541b1a0d",
        "reply": Object {
          "id": "r1",
          "text": "
      「外國同性戀藉結婚來台治療愛滋病，都是健保支付，全民買單」這句話其實並不算錯，但實務上並不會有「拖垮健保」的狀況，也跟同性婚姻關係不大，所以不用太擔心唷。
      衛福部 - 有關網路流傳外國人染愛滋病來台2年後，所有外國人的愛滋病治療費用是由健保署負擔，疾管署特此說明
      https://www.cdc.gov.tw/info.aspx?treeid=45da8e73a81d495d&nowtreeid=1bd193ed6dabaee6&tid=14D38681253864AB&fbclid=IwAR1mtdGIYYbHdiB99pZg_YQSwGXUD-DMj47fmgFnrphQ778k73SQzZkEUDk
    ",
        },
      },
    ],
  },
}
`;

exports[`Basic object type graph resolving loads from reply to article 1`] = `
Object {
  "reply": Object {
    "hyperlinks": Array [
      Object {
        "title": "有關網路流傳外國人染愛滋病來台2年後，所有外國人的愛滋病治療費用是由健保署負擔，疾管署特此說明 ",
      },
    ],
    "paragraphReplies": Array [
      Object {
        "canDelete": true,
        "id": "5bd677bb092a3ee2541b1a0d",
        "paragraph": Object {
          "article": Object {
            "id": "article1",
          },
          "canDelete": true,
          "id": "p1",
        },
      },
    ],
  },
}
`;
